orc8r/cloud/go/obsidian/access/rest_middleware.go

```diff
@@ -34,83 +34,69 @@ import (
 
 func Middleware(next echo.HandlerFunc) echo.HandlerFunc {
 	return func(c echo.Context) error {
-		if c == nil || c.Request() == nil {
-			return handleError(c, http.StatusBadRequest, "Invalid Request")
+		req := c.Request()
+		if req == nil {
+			return makeErr(c, http.StatusBadRequest, "invalid request")
 		}
-		glog.V(1).Infof("Received request in the access middleware. Req: %v", c.Request())
-		// find out request's access type (READ|WRITE|READ & WRITE)
-		perm := requestPermissions(c)
 
-		// Get Request's Operator
-		oper, err := RequestOperator(c)
+		glog.V(1).Infof("Received request in access middleware: %+v", req)
+
+		operator, err := getOperator(c, req)
 		if err != nil {
-			if _, ok := err.(errors.ClientInitError); ok {
-				return handleError(c, http.StatusServiceUnavailable, "Service Unavailable")
-			}
-			return handleError(
-				c,
-				http.StatusUnauthorized,
-				"Client Credentials Error: %s", err)
+			return transformErr(err, c, http.StatusUnauthorized, "Invalid client credentials: %s", err)
 		}
-		if oper == nil {
-			return handleError(
-				c,
-				http.StatusUnauthorized,
-				"Missing Client Credentials")
+		if operator == nil {
+			return makeErr(c, http.StatusUnauthorized, "missing client credentials")
 		}
 
-		// Bypass farther identity Checks for static docs GET having an
-		// operator cert should be enough
-		if urlPath := c.Path(); perm != accessprotos.AccessControl_READ || !(strings.HasPrefix(urlPath, obsidian.StaticURLPrefix)) {
+		perms := getRequestedPermissions(c, req)
+		staticReadOnly := perms == accessprotos.AccessControl_READ && strings.HasPrefix(c.Path(), obsidian.StaticURLPrefix)
+		if !staticReadOnly {
 			// Get Request's Entities' Ids
 			ids := FindRequestedIdentities(c)
 
 			// Check Operator's ACL for required entity permissions
-			ents := make([]*accessprotos.AccessControl_Entity, len(ids))
-			for i, e := range ids {
-				ents[i] = &accessprotos.AccessControl_Entity{Id: e, Permissions: perm}
+			var ents []*accessprotos.AccessControl_Entity
+			for _, id := range ids {
+				ents = append(ents, &accessprotos.AccessControl_Entity{Id: id, Permissions: perms})
```

Comment:
This change is unnecessary and less efficient then original, please remove it.

Reply:
Unless I'm misunderstanding, the len(ents) should be on the order of 1, or at most 10? If so, this change would be a performance hit on the order of less than a microsecond per request

I'll revert if you feel strongly, just lmk

Reply:
I wasn't referring to use of len(), I was talking about your use of append() inside the loop.
It's not efficient: the original implementation pre-allocates the result array one time and "fills" it in with ACL entities,
your implementation allocates an empty array (slice) and then potentially reallocates and copies it on every loop iteration. Optimizations in the compiler's append() implementation will probably mitigate some of the reallocation inefficiencies, but not all, so - it's not a good practice in general.

Reply:
Right no I understand. I'm just saying the performance hit is trivial and not worth considering. But I can revert since I don't feel that strongly